src/middleware/withLocator.js

```diff
@@ -46,3 +44,22 @@ export function withLocator (handler) {
     return handler(request, env, { ...ctx, locator })
   }
 }
+
+/**
+ * Determines if the indexing service is enabled. It is enabled if the request
+ * contains the `ff=indexing-service` query parameter or if a random chance
+ * falls within the ramp-up percentage. If `FF_RAMP_UP_PERCENTAGE` is not set,
+ * it defaults to 0%.
+ *
+ * @param {Request} request
+ * @param {LocatorEnvironment} env
+ * @returns {boolean}
+ */
+function isIndexingServiceEnabled (request, env) {
+  const withIndexingServicesArg = new URL(request.url).searchParams
+    .getAll('ff')
+    .includes('indexing-service')
+  const percentage = env.FF_RAMP_UP_PERCENTAGE ? Number(env.FF_RAMP_UP_PERCENTAGE) : 0.0
```

Comment:
**suggestion:** Let's either make `FF_RAMP_UP_PERCENTAGE` be 0–100, or change the name to not use "percentage"—maybe "probability"? I think the clearest is probably to use 0–100. We usually talk about it as a percentage, so we'll be setting it "to 50%", rather than "to 0.5".

Reply:
sounds good @Peeja . I will update that. Thanks